src/internal/rawdataextractcallbackwid.cpp

```diff
@@ -0,0 +1,70 @@
+// This is an open source non-commercial project. Dear PVS-Studio, please check it.
+// PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+
+/*
+ * bit7z - A C++ static library to interface with the 7-zip shared libraries.
+ * Copyright (c) 2014-2024 Riccardo Ostani - All Rights Reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+#include "rawdataextractcallbackwid.hpp"
+
+#include "internal/rawdataextractcallback.hpp"
+
+#include "bitinputarchive.hpp"
+#include "bitpropvariant.hpp"
+#include "internal/crawoutstream.hpp"
+#include "internal/extractcallback.hpp"
+#include "internal/util.hpp"
+
+#include <cstdint>
+#include <ostream>
+#include <utility>
+
+using namespace NWindows;
+
+namespace bit7z {
+
+RawDataExtractCallbackWid::RawDataExtractCallbackWid( const BitInputArchive& inputArchive, CallbackType& callback )
```

Comment:
todo 

once you approve of idea I will unify it with RawDataExtractCallback class